core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -281,6 +283,9 @@ public class SqlToRelConverter {
    */
   private final Deque<String> datasetStack = new ArrayDeque<>();
 
+  /** Stack of currently processed operators. */
```

Comment:
This code comment could be more detailed,you could give a code comment as to why this stack is there,it is better to help others understand.